dcpy/library/archive.py

```diff
@@ -119,19 +119,21 @@ def __call__(
                 # Find all files in latest where the version (stored in s3 metadata)
                 # does not match the version of the file currently getting added to latest
                 keys_in_latest = self.s3.ls(f"datasets/{name}/latest")
-                if len(keys_in_latest) > 0:
-                    diff_version = [
-                        k
-                        for k in keys_in_latest
-                        if self.s3.info(k)["Metadata"].get("version", "") != version
-                    ]
-
-                    # Remove keys from the latest directory that have versions different
-                    # from the version of the file currently getting added to latest
+                diff_version = [
+                    k
+                    for k in keys_in_latest
+                    if self.s3.info(k)["Metadata"].get("version", "") != version
+                ]
+
+                # Remove keys from the latest directory that have versions different
+                # from the version of the file currently getting added to latest
+                if len(diff_version) > 0:
```

Comment:
I feel like this hasn't been working? Haven't we had differing versions in latest folders